### DISCUSSION TITLE
Combine keybinding actions with binary or (`||`) operator to stop executing on first successful lazy function

### DISCUSSION THREAD

**Discussion by sahinakkaya** [qtile/qtile#3923] [Ideas] [answered]
I want to use single keybinding with multiple layouts. My keybindings look like this:
```
Key([mod], "h", lazy.layout.left(), lazy.layout.previous(), desc="Move focus to the left / previous"),
```
This will work as intended if only one of the actions is available for a layout (i.e. only left or only previous). But if both of them available for a layout `mod + h` triggers both of them and the focus changes twice. 

To solve this issue, I created a custom function which executes the actions until the first successful run.

```
def run_one_of(instance, *args):
    for arg in args:
        try:
            getattr(instance, f"cmd_{arg}")()
        except AttributeError:
            pass
        else:
            break

...
Key([mod], "h", lazy.layout.function(run_one_of, "left", "previous"), 
        desc="Move focus to left / previous"),
```

I want to use this function to "`maximize` a *layout*" or "`toggle_fullscreen` for a *window*". Since these are different instances (layout and window), I can't do something like this:
```
    Key([mod], "m", lazy.layout.function(run_one_of, "maximize", "toggle_fullscreen", desc="Maximize layout / toggle fullscreen"),
                      #  ^^^^ it should be window for toggle_fullscreen
```

And here is my idea: if it was possible to combine lazy functions with `||`, we could achieve the same result elegantly. The `||` operator would run the lazy function sequence until the first successful run, just like the bash `||` operator. 

So the above keybindings would look like:

```
Key([mod], "h", lazy.layout.left() || lazy.layout.previous()
    desc="Move focus to the left / previous"),
Key([mod], "m", lazy.layout.maximize() || lazy.window.toggle_fullscreen(), 
    desc="Maximize layout / toggle fullscreen"),
```

What do you think about this?

**Comment by elParaguayo** [comment, marked as answer]
lazy commands can take a `.when` suffix where you can specify which layouts the command should be run on e.g. `.when(layout="monadtall")` or `.when(layout=["monadtall", "monadwide"])`.


Would that solve your issue?

**Reply by sahinakkaya** [reply]
I need to read the docs carefully :smile:  Thanks, that solves the problem.
